src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -291,6 +293,21 @@ def children_plan_nodes(self) -> List[Union["Selectable", "SnowflakePlan"]]:
         else:
             return []
 
+    def replace_child(self, old_node, new_node) -> None:
+        """This method is called during optimization stage to cut plan tree at a certain node.
```

Comment:
@sfc-gh-aalam here is something kind of bothers me, for regular logical plans like limit, snowflakeTables, the update seems fine to me, because the node is still valid. However, for resolved nodes like SnowflakePlan and Selectable, after the update the node will not be valid anymore, for example, the queries, post actions may change, provide such general method on the node its-self doesn't seem something safe, and when user will need to pay attention when use this method.

I am thinking shall we just move the replace child as one single utility function, and do it like
```
def replace_child(parent, child, new_child)
    if isinstance(parent, SelectStatement):
         xxxx
    if isintance(xxxx):
         xxxx 
```
and only use this during our compilation phase. The disadvantage is that if we added a new  logical node, we might miss the correct way for update if the developer didn't pay attention to it, but i think that issue can still remains as far as we already provided a default way to update the  node.